File: packages/domain/lib/src/repositories/skill_repository.dart
import 'package:fmf_domain/src/models/progress_snapshot.dart';
import 'package:fmf_domain/src/models/skill.dart';

abstract interface class SkillRepository {
  Future<List<Skill>> getSkills();
  Future<Skill?> getSkillById(String id);
  Stream<ProgressSnapshot?> watchSkillProgress(String skillId);
}

File: apps/ios_app/Sources/Domain/Repositories/SkillRepository.swift
import Foundation

protocol SkillRepository: Sendable {
    func getSkills() async throws -> [Skill]
    func getSkillById(_ id: String) async throws -> Skill?
    func skillProgressStream(skillId: String) -> AsyncStream<ProgressSnapshot?>
    func getProgressSnapshot(skillId: String) async throws -> ProgressSnapshot?
    func saveProgressSnapshot(_ snapshot: ProgressSnapshot) async throws
}
